### PR TITLE
docs: add jsdoc & add usage in README

### PR DESCRIPTION
**Pull request by whtsky**

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -9,9 +9,19 @@ export interface IParseInput {
...
+  /**
+   * The name of the function used to determine the order of object keys.
+   * If omitted, the keys are sorted in ascending, ASCII character order.
+   * Pass `false` to disable sorting.
+   */
   sort?: false | ((a: string, b: string) => number);
 }
 
+/**
+ * Gets the unencoded version of an encoded component of a Uniform Resource Identifier (URI).
```
Comment (frantic1048) [inline index.ts]: How about "... of an encoded URI component" ?

File: index.ts [hunk 2]
```diff
@@ -20,6 +30,10 @@ function decodeUriComponent(input: string) {
   }
 }
 
+/**
+ * Encodes a text string as a valid component of a Uniform Resource Identifier (URI)(RFC3986)
```
Comment (frantic1048) [inline index.ts]: The same as above.

**Review by frantic1048 — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -8,5 +8,30 @@ Parse and stringify URL query strings. Forked from [query-string](https://github
 yarn add @chaitin/querystring
 ```
 
+## Usage
+```ts
+import { parse, stringify } from '@chaitin/querystring'
+
+const result = parse("foo=foo&foo&foo=one&foo=&bat=buz")
+// the result is:
+{ bat: "buz", foo: ["foo", null, "one", ""] }
```
Comment (frantic1048) [inline README.md:17]: Where are our moe moe friends? (

File: index.ts [hunk 4]
```diff
@@ -84,6 +102,10 @@ export function parse(input: string): IParseReturn {
   return ret;
 }
 
+/**
+ * Produces a URL query string from a given obj
+ * @param obj the object to be stringified
+ */
```
Comment (KagamiChan) [inline index.ts]: seems the 2nd argument, `options` should be listed

could its definition be read when hovering on the function name in VSC?

**Review by KagamiChan — CHANGES_REQUESTED** [review]

File: index.ts (hunk 4 above)
Comment (whtsky) [reply index.ts]: > could its definition be read when hovering on the function name in VSC?

nop

File: README.md (hunk 3 above)
Comment (whtsky) [reply README.md:17]: PR please (

**Comment by whtsky** [comment]
updated

**Review by KagamiChan — APPROVED** [review]
LGTM

**Review by frantic1048 — APPROVED** [review]
LGTM

**Pull request merged**